### PR TITLE
Add /namecolors command

### PR DESCRIPTION
**Pull request by OTCompa**
Requested by @clairefleuret
Adds a command to give the invoker the respective name color role as configured in `config.yaml`. Only one name color role will be applied at a given time and will remove any other name color roles already present on the user. Can be used again to toggle off the role.

## config.yaml
Based on what's already in `encounters.go`.
To enable the command in a guild: `nameColor` must be set to `true` in the guild's config.
Role `type` should be `Name Color`
Here's a sample `config.yaml` which should work for the examples below.
```yaml
guilds:
- name: NAUR
  roles:
    nameColor: true
  guildId: 1234567890
  channelId: 1234567890
  encounters:
  - ids: [1060, 1047, 1039, 1073]
    name: "The Unending Coil of Bahamut (Ultimate)"
    defaultRoles: false
    totalWeaponsAvailable: 15
    the: "Legendary"
    roles:
      - name: "UCoB Cleared"
        type: "Cleared"
        color: 0xfdfd96
      - name: "The Legend's Color"
        type: "Name Color"
        color: 0xfdfd96
```
A separate PR will be made to modify `config.yaml` so that the commands will be set up for NAUR.
## Examples:
(Ignore the bot name I just reused the same token lol)
image

If prerequisite role is not present:
image

If prerequisite role is present and no other name color role is present:
image

If prerequisite role is present and another name color role is present:
image

If the same name color role is present (toggle role):
image

### CODE REVIEW

**Comment by ghost** [comment]
All use cases have been tested, no issues found.

File: internal/clearingway/handlers.go [hunk 1]
```diff
@@ -174,6 +182,47 @@ var ProgCommand = &discordgo.ApplicationCommand{
...
+					Name: "DSR",
+					Value: "Dragonsong's Reprise (Ultimate)",
+				},
+				{
+					Name: "TOP",
+					Value: "The Omega Protocol (Ultimate)",
+				},
+				/*
+				{
+					Name: "FRU",
```
Comment (ghost) [inline internal/clearingway/handlers.go:244]: Prepared for FRU release, can be uncommented when the fight goes live.

File: internal/clearingway/handlers.go [hunk 2]
```diff
@@ -466,6 +517,103 @@ func (c *Clearingway) RemoveAll(s *discordgo.Session, i *discordgo.InteractionCr
...
+	g, ok := c.Guilds.Guilds[i.GuildID]
+	if !ok {
+		fmt.Printf("Interaction received from guild %s with no configuration!\n", i.GuildID)
+		return
+	}
+
+	// Ignore messages not on the correct channel
+	if i.ChannelID != g.ChannelId {
+		fmt.Printf("Ignoring message not in channel %s.\n", g.ChannelId)
+	}
```
Comment (brtran4) [inline internal/clearingway/handlers.go]: just like the first PR, let's add an ephemeral message to the user to use it in the correct channel

File: internal/clearingway/handlers.go [hunk 3]
```diff
@@ -466,6 +517,103 @@ func (c *Clearingway) RemoveAll(s *discordgo.Session, i *discordgo.InteractionCr
...
+		tempstr := ""
+		if roleToRemove != nil {
+			fmt.Printf("Removing role: %+v\n", roleToRemove.Name)
+			err = roleToRemove.RemoveFromCharacter(g.Id, i.Member.User.ID, c.Discord.Session)
+			if err != nil {
+				fmt.Printf("Error removing role: %+v\n", err)
+				return
+			}
+			tempstr += fmt.Sprintf("Successfully removed role: <@&%v>", roleToRemove.DiscordRole.ID)
+		}
```
Comment (brtran4) [inline internal/clearingway/handlers.go]: i'm kind of surprised that there's no helper function to begin with (before your PRs). let's make this a helper

File: internal/clearingway/handlers.go [hunk 4]
```diff
@@ -466,6 +517,103 @@ func (c *Clearingway) RemoveAll(s *discordgo.Session, i *discordgo.InteractionCr
...
+		// add role if requested role is not the same as color role
+		if requestedColorRole != roleToRemove {
+			fmt.Printf("Adding role: %+v\n", requestedColorRole.Name)
+			err = requestedColorRole.AddToCharacter(g.Id, i.Member.User.ID, c.Discord.Session)
+			if err != nil {
+				fmt.Printf("Error adding role: %+v\n", err)
+				return
+			}
+			tempstr += fmt.Sprintf("\nSuccessfully added role: <@&%v>", requestedColorRole.DiscordRole.ID)
```
Comment (brtran4) [inline internal/clearingway/handlers.go]: this also goes with making this a helper as well. we might as well replace all current implementations of adding a role with this helper in a separate PR. but, for now let's separate this into a helper and replace only this piece of code with the helper (and other places you've written similar code)

File: internal/clearingway/handlers.go [hunk 5]
```diff
@@ -466,6 +517,103 @@ func (c *Clearingway) RemoveAll(s *discordgo.Session, i *discordgo.InteractionCr
...
+		// edge case where someone has clears removed and doesn't want the color 
+		if requestedColorRole == roleToRemove {
+			fmt.Printf("Removing role: %+v\n", roleToRemove.Name)
+			err = roleToRemove.RemoveFromCharacter(g.Id, i.Member.User.ID, c.Discord.Session)
+			if err != nil {
+				fmt.Printf("Error removing role: %+v\n", err)
+				return
+			}
+			tempstr := fmt.Sprintf("Successfully removed role: <@&%v>", roleToRemove.DiscordRole.ID)
+			discord.ContinueInteraction(s, i.Interaction, tempstr)
```
Comment (brtran4) [inline internal/clearingway/handlers.go]: same concept here

**Review by brtran4 — CHANGES_REQUESTED** [review]

**Review by brtran4 — APPROVED** [review]

**Pull request merged**